element_array_ephys/ephys_acute.py

```diff
@@ -662,6 +664,74 @@ def yield_unit_waveforms():
             self.Waveform.insert(unit_electrode_waveforms, ignore_extra_fields=True)
 
 
+@schema
+class QualityMetric(dj.Imported):
+    definition = """
+    # Clusters and waveforms metrics
+    -> CuratedClustering    
+    """
+
+    class Cluster(dj.Part):
+        definition = """   
+        # Cluster metrics for a particular unit
+        -> master
+        -> CuratedClustering.Unit
+        ---
+        firing_rate=null: float # (Hz) firing rate for a unit 
+        snr=null: float  # signal-to-noise ratio for a unit
+        presence_ratio=null: float  # fraction of time in which spikes are present
+        isi_violation=null: float   # rate of ISI violation as a fraction of overall rate
+        number_violation=null: int  # total number of ISI violations
+        amplitude_cutoff=null: float  # estimate of miss rate based on amplitude histogram
+        isolation_distance=null: float  # distance to nearest cluster in Mahalanobis space
+        l_ratio=null: float  # 
+        d_prime=null: float  # Classification accuracy based on LDA
+        nn_hit_rate=null: float  # Fraction of neighbors for target cluster that are also in target cluster
+        nn_miss_rate=null: float # Fraction of neighbors outside target cluster that are in target cluster
+        silhouette_score=null: float  # Standard metric for cluster overlap
+        max_drift=null: float  # Maximum change in spike depth throughout recording
+        cumulative_drift=null: float  # Cumulative change in spike depth throughout recording 
+        contamination_rate=null: float # 
+        """
+
+    class Waveform(dj.Part):
+        definition = """   
+        # Waveform metrics for a particular unit
+        -> master
+        -> CuratedClustering.Unit
```

Comment:
Are these metrics calculated for the [peak_electrode_waveform](https://github.com/datajoint/element-array-ephys/blob/ee0e179d0ed02212f03a36382a07409b6ba2f823/element_array_ephys/ephys_acute.py#L573), [waveform_mean](https://github.com/datajoint/element-array-ephys/blob/ee0e179d0ed02212f03a36382a07409b6ba2f823/element_array_ephys/ephys_acute.py#L583), or [waveforms](https://github.com/datajoint/element-array-ephys/blob/ee0e179d0ed02212f03a36382a07409b6ba2f823/element_array_ephys/ephys_acute.py#L584)?

Reply:
Based on the design, I presume it is the `peak_electrode_waveform`, but confirming.

Reply:
Depending on the metric, but the calculation is based off of `peak_electrode_waveform` and also `waveforms` (some metrics require looking at the waveform propagation across channels, not just the one from the peak channel)

Reply:
Thanks